src/server/saved_objects/service/lib/repository.js

```diff
@@ -139,6 +148,8 @@ export class SavedObjectsRepository {
     } = options;
     const time = this._getCurrentTime();
     const bulkCreateParams = [];
+    objects = objects.filter(object => this._isTypeAllowed(object.type));
```

Comment:
Consumers of `bulk_create` expect to have a response for each object they attempt to create. Filtering out types here could result in the `saved_objects` response having less entries than they sent in.

I think we should return an inline error for types that aren't allowed, which is consistent with what we do for errors returned by ES here. Something like:

```
saved_objects: [{
  id: 'my-dashboard',
  type: 'dashboard',
  updated_at: 123456,
  version: 1,
  attributes: { ... },
  references: [ ... ]
}, {
  id: 'some-bogus-document',
  type: 'some-bogus-type',
  error: {
    "statusCode": 400,
    "message": "some error message here" 
  }
}, {
  id: 'my-visualization',
  type: 'visualization',
  updated_at: 123456,
  version: 1,
  attributes: { ... },
  references: [ ... ]
}]

```